CMakeLists.txt

```diff
@@ -1302,7 +1303,43 @@ if(BUILD_NVFUSER_BENCHMARK)
     add_dependencies(nvfuser_multidevice_bench flatc build_flatbuffer_config)
 
     if(NOT MSVC)
-      target_compile_options(nvfuser_bench PRIVATE
+      target_compile_options(nvfuser_multidevice_bench PRIVATE
+        -Wall -Wno-unused-function
+        -Werror -Wno-deprecated-copy
+      )
+    endif()
+
+    # cuda ipc benchmark
+    set(P2P_COMMUNICATION_BENCHMARK_SRCS)
+    list(APPEND P2P_COMMUNICATION_BENCHMARK_SRCS
+      ${NVFUSER_ROOT}/benchmarks/cpp/p2p_communication.cpp
+      ${NVFUSER_ROOT}/tests/cpp/utils.cpp
+    )
+    add_executable(nvfuser_p2p_communication_bench ${P2P_COMMUNICATION_BENCHMARK_SRCS})
+    set_target_properties(nvfuser_p2p_communication_bench PROPERTIES
+      C_STANDARD ${NVFUSER_C_STANDARD}
+      CUDA_STANDARD ${NVFUSER_CUDA_STANDARD}
+      CXX_STANDARD ${NVFUSER_CPP_STANDARD}
+      CXX_STANDARD_REQUIRED ON
+      CXX_VISIBILITY_PRESET hidden
+      POSITION_INDEPENDENT_CODE Yes
+      VISIBILITY_INLINES_HIDDEN Yes
+    )
+    target_include_directories(nvfuser_p2p_communication_bench SYSTEM PRIVATE
+      ${CMAKE_SOURCE_DIR}/third_party/benchmark/include
+      ${CMAKE_SOURCE_DIR}/third_party/flatbuffers/include
+      ${CMAKE_SOURCE_DIR}/third_party/googletest/googletest/include
```

Comment:
I don't think your p2p_communication.cpp use these headers at all. It's really just an ad-hoc binary, not depending on the Google benchmark infrastructure.